File: lib/core/models/user_model.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';

part 'user_model.freezed.dart';
part 'user_model.g.dart';
@freezed
class UserModel with _$UserModel {

  const factory UserModel({
    @JsonKey(name: '_id') required String uid,
    required String email,
    required String displayName,
    String? password,
    int? genderId,
    String? token,
    required DateTime createdAt,
    required DateTime lastSeen,
    String? photoProfile,
    required String role,
    DateTime? dateBirth,
    String? phone,
    String? countryCode,
    String? countryName,
    String? dialingCode,
    required bool isVerified,
    required bool isBanned,
    required bool hideIntro,
    required List<String> blocks,
    String? bio
  }) = _UserModel;


  factory UserModel.fromJson(Map<String, Object?> json)
  => _$UserModelFromJson(json);
}

File: functions/database/models/user_model.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;



const userSchema = Schema(
    {
        '_id': {
            type: String,
            required: true,
        },
        'email': {
            type: String,
            required: true,
        },
        'displayName': {
            type: String,
            required: true,
        },
        'password': {
            type: String,
            required: false,
        },
        'genderId': {
            type: Number,
            required: false,
        },
        'token': {
            type: String,
            required: false,
        },
        'lastSeen': {
            type: Date,
            required: true,
        },
        'photoProfile': {
            type: String,
            required: false,
        },
        'role': {
            type: String,
            required: true,
        },
        'dateBirth': {
            type: Date,
            required: false,
        },
        'phone': {
            type: String,
            required: false,
        },
        'countryCode': {
            type: String,
            required: false,
        },
        'countryName': {
            type: String,
            required: false,
        },
        'dialingCode': {
            type: String,
            required: false,
        },
        'isVerified': {
            type: Boolean,
            required: true,
        },
        'isBanned': {
            type: Boolean,
            required: true,
        },
        'hideIntro': {
            type: Boolean,
            required: true,
        },
        'bio': {
            type: String,
            required: false,
        },
        'createdAt': {
            type: Date,
            default: Date.now,
        },
        'updatedAt': {
            type: Date,
            default: Date.now,
        },
        'friends': [{
            type: String,
            ref: 'Friend'
        }],
        'blocks': [{
            type: String,
            ref: 'User'
        }],
        'runtimeType': {
            type: String,
            required: false,
        },
    },
    {
        timestamps: true ,
        collection: 'users',
        toObject: {
            transform: function (doc, ret, options) {
                delete ret.__v;
            },
        },
    },
);

module.exports = mongoose.model('User', userSchema);
